floodsub.go

```diff
@@ -70,11 +70,20 @@ func (fs *FloodSubRouter) Publish(from peer.ID, msg *pb.Message) {
 			continue
 		}
 
+		if out.Size() > maxRPCSize {
```

Comment:
Is this check really necessary here?
We already drop large messages on publish and can't read messages larger than maxRCPSize anyway.

Reply:
I guess it should be fine in floodsub. My worry is gossipsub.

Reply:
well, let's remove it from here as it's redundant, and rethink the gossipsub case.

Reply:
This can't hurt and future proofs us against, e.g., batching too many messages in a single RPC and then silently failing when the remote side drops it.